gammapy/maps/wcsnd.py

```diff
@@ -495,38 +498,116 @@ def plot(self, ax=None, fig=None, add_cbar=False, stretch='linear', **kwargs):
 
-        if not self.geom.is_image:
-            raise TypeError('Only supported on 2D maps')
 
         if fig is None:
             fig = plt.gcf()
 
-        if ax is None:
-            ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
-
-        data = self.data
 
         kwargs.setdefault('interpolation', 'nearest')
         kwargs.setdefault('origin', 'lower')
         kwargs.setdefault('cmap', 'afmhot')
-        norm = simple_norm(data[np.isfinite(data)], stretch)
-        kwargs.setdefault('norm', norm)
 
-        caxes = ax.imshow(data, **kwargs)
 
-        cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
-        try:
-            ax.coords['glon'].set_axislabel('Galactic Longitude')
-            ax.coords['glat'].set_axislabel('Galactic Latitude')
-        except KeyError:
-            ax.coords['ra'].set_axislabel('Right Ascension')
-            ax.coords['dec'].set_axislabel('Declination')
-        except AttributeError:
-            log.info("Can't set coordinate axes. No WCS information available.")
-
-        # without this the axis limits are changed when calling scatter
-        ax.autoscale(enable=False)
-        return fig, ax, cbar
+
+        if self.geom.ndim == 2:
+            if ax is None:
+                ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
+
+            data = self.data
+            norm = simple_norm(data[np.isfinite(data)], stretch)
+            kwargs.setdefault('norm', norm)
+            caxes = ax.imshow(data, **kwargs)
+            cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
+            # without this the axis limits are changed when calling scatter
+            ax.autoscale(enable=False)
+
+            try:
+                ax.coords['glon'].set_axislabel('Galactic Longitude')
+                ax.coords['glat'].set_axislabel('Galactic Latitude')
+            except KeyError:
+                ax.coords['ra'].set_axislabel('Right Ascension')
+                ax.coords['dec'].set_axislabel('Declination')
+            except AttributeError:
+                log.info("Can't set coordinate axes. No WCS information available.")
+
+            return fig, ax, cbar
+
+        elif self.geom.ndim == 3:
+
+            try:
+                from ipywidgets.widgets.interaction import interact, fixed
+                import ipywidgets as widgets
+            except:
+                raise ImportError('ipywidgets is not installed.')
+
+            @interact(
+                     index=widgets.IntSlider(min=0, max=self.data.shape[0] - 1, step=1, value=1,
+                                             description=self.geom.axes_names[0]+' slice'),
+                     stretch=widgets.RadioButtons(options=['linear', 'sqrt', 'log'], value='sqrt',
+                                                  description='Plot stretch'),
+                     mapND=fixed(self)
+                    )
+            def _plot_interactive(mapND, index, stretch='linear'):
```

Comment:
It would be great to still pass `**kwargs` to the `ax1.imshow()` command below.

Reply:
I've been trying a few things and searching the internet and could not find a working example. the thing is that for each parameter in @interact you have to specify whether it will be fixed or interactive. I don't how @interact handles optional keywords.

Reply:
The `kwargs` collected in `.plot()` is just a plain python dict, that can be passed as a fixed parameter to the `_plot_interactive()` method without the `**` operator, here is a minimal example:
```python
def h(c):
    print(c)

def f(a, **kwargs):
    print(a)
    def _g(b, kwargs):
        print(b)
        h(**kwargs)   
    _g('arg b', kwargs)
``` 
This should work, no?

Reply:
yes I just implemented that and it worked thanks.